### ISSUE TITLE
Is the GPL3 license too restrictive

### ISSUE THREAD

**Issue by smoyer64** [MichaelMure/git-bug#874]
Since the entities and caching are being generalized to allow others to use `git-bug` as a library for storing data in `git`, is the GPL3 license too restrictive?  Will needing to release derivative products under the same GPL3 license dissuade people from adopting `git-bug` as a library?  Can we build a bigger eco-system with a less restrictive license?

**Comment by MichaelMure** [comment]
Possibly ... I don't have strong opinion on the license. However, as far as I understand, changing the license for some packages would require approval of all authors, as well as making sure that only compatible licenses are used for the dependency. This is no small task ... Just listing dependencies and authors is a pain apparently.

**Comment by smoyer64** [comment]
To change the license of `git-bug`, we would indeed need to track down the authors of code in `git-bug`'s packages but at least we wouldn't have to track down the authors of the dependencies!

Perhaps the sane thing to do is to close this issue and wait for someone to complain (although we'll never know if someone is considering `git-bug` as a library and rejects it due to the license).  It's not an issue for `git-bug` as an application because there are no dependencies with licenses that are MORE restrictive.

**Comment by MichaelMure** [comment]
Another problem with MIT for the core and keep GPL for the outer layers (entities, CLI, bridges ..) is that code routinely move from one to another (ex: prototyping in entities then generalizing in the core). Things get blurry fast, even if I'm the author of the vast majority of the core.

**Comment by smoyer64** [comment]
Agreed - if you look at the file headers for each source file in the Linux repository, you'll see that there are one to many copyright holders listed - that seems like too much of an administrative burden (e.g. https://github.com/torvalds/linux/blob/master/fs/binfmt_flat.c.)

**Comment by MichaelMure** [comment]
We could do something similar as what Protocol Labs did: https://github.com/ipfs/kubo/issues/6302
- collect past contributor's approval
- switch and hope for the best (?)

**Comment by smoyer64** [comment]
Since the ability to use `git-bug` as "a database" is new with the recent refactorings, it's hard to identify a community of those that would benefit from a different license (or perceive the existing license as being too restrictive).  I guess polling the current committers for their opinions would be a place to start? But ...

> switch and hope for the best (?)

As you noted above, this represents quite a bit of work and since there are currently no CLAs in place, what's allowed is somewhat undefined (beyond continuing the project without changes.)  How do we get feedback on this?  And if nobody cares about the derivative works clauses, the original point is moot!
